### DISCUSSION TITLE
Adding a newline to a variable

### DISCUSSION THREAD

**Discussion by nobullituesday** [community/community#122707] [Actions] [answered]
### Select Topic Area

Question

### Body

So I'm a novice programmer in training trying to implement GitHub Actions for a repository. As of right now I got most of it working. The general idea is that I would like to automatically check PRs if our guidelines have been violated. In this specific case I need to check if more than one angular component has been worked on in a single PR and accordingly post a sticky comment to the PR. 

However I am running into trouble when I try to add the names of the changed files to the sticky comment since I would like them to be separated by a newline.

I've tried delcaring the newline as a variable, enconding it as %0A (like it is in the example below) and many different variations. But I can't seem to properly add the newline. The closest I've gotten so far is having a \n between the file names without them actually being separated by a newline.

In the example below i want the following behavior in my for loop:

changed_components+= "$file" +NEWLINE 

`
        run: |
          changed_components=""
          for file in ${ALL_CHANGED_FILES}; do
            if echo "$file" | grep -q "component"; then
              changed_components+="$file"%0A
            fi
          done
          echo -e "CHANGED_COMPONENT_FILES=${changed_components}" >> $GITHUB_ENV
`

Any help would be greatly appreciated!

**Comment by ChristopherHX** [comment, marked as answer]
The github env output file `$GITHUB_ENV` doesn't support the old escaping rules of `::set-env::`

https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings

You should preserve newlines as is and not encode them, wrap them like in the example in the docs e.g.
```
{name}<<{delimiter}
{line1}
{line2}
{delimiter}
```

Don't use the `=` for multiline env / outputs. And remove replace the placeholder in brackets

**Reply by nobullituesday** [reply]
This helped greatly! 

I've actually stumbled across this already but did not quite understand it in the docs. However with your input, knowing that it is the right solution, I managed to get it working.

Thanks a lot
